TSPL.docc/LanguageGuide/Concurrency.md

```diff
@@ -564,33 +637,192 @@ the explicit parent-child relationships between tasks
 let Swift handle some behaviors like propagating cancellation for you,
 and lets Swift detect some errors at compile time.
 
+[`TaskGroup`]: https://developer.apple.com/documentation/swift/taskgroup
+
+<!-- TR: What's a good example of this kind of error? -->
```

Comment:
It's more about "omission problems" rather than actual "errors"

So for example structured concurrency automatically:
- when parent is cancelled, child task is cancelled -- doing this manually is very difficult
- task priority -- when we have a chain of tasks including child tasks priority gets escalated efficiently through child tasks
- task local values are propagated efficiently and automatically to child tasks
- you CANNOT forget to wait for a child task; structured concurrency prevents "task (number) explosions" because you always wait for them to complete


Doing all this by hand is a world of pain; so this is mostly about this

We did a WWDC talk on this this year; here's short notes from it: https://www.wwdcnotes.com/notes/wwdc23/10170/

And here's the whole talk: https://developer.apple.com/videos/play/wwdc2023/10170


Could be worth linking if we can